src/lib.rs

```diff
@@ -25,15 +25,14 @@
 //! use std::path::Path;
```

Comment:
It is possible to wrap this whole test in a hidden `#[cfg(…)]` even in a doctest (note the preceding `#`). That should exclude the block when the `encoder` feature is not enabled and allow you to use `no_run` again.

```
//! # #[cfg(feature = "encoder")] {
…
//! # }
```